src/cosim/observer/time_series_observer.cpp

```diff
@@ -263,10 +274,10 @@ void time_series_observer::simulator_step_complete(
 }
 
 void time_series_observer::state_restored(
-    step_number currentStep, time_point currentTime)
+    step_number, time_point)
 {
     for (const auto& entry : slaveObservers_) {
-        entry.second->observe(currentStep, currentTime);
+        entry.second->reset();
```

Comment:
I experienced an issue with time-series-observer when the state was restored.

The observer was not able to fetch correct time-series values via `get_step_numbers` and `get_real_samples` when the state was restored back in simulation time. This was due to the earlier simulation data in buffer `timeSamples_` was cleared (`adjustIfFull`). Because of this, the data needed in the restored state was not available anymore and the observer was no longer able to retrieve the correct time series data unless the simulation progresses until the time matches those in "max(`timeSamples_`)".

Since the data will be "re-generated" anyways after the restore, I suggest just to clear the buffers (both `timeSamples_` and `step_numbers`) to avoid this issue.